client/lib/plans/features-list.js

```diff
@@ -708,44 +708,44 @@ export const FEATURES_LIST = {
 	},
 	[ constants.FEATURE_BACKUP_ARCHIVE_30 ]: {
 		getSlug: () => constants.FEATURE_BACKUP_ARCHIVE_30,
-		getTitle: () => i18n.translate( '30-day Backup Archive' ),
+		getTitle: () => i18n.translate( '30-day backup archive' ),
 		getDescription: () =>
 			i18n.translate( 'Browse or restore any backup made within the past 30 days.' ),
 	},
 	[ constants.FEATURE_BACKUP_ARCHIVE_15 ]: {
 		getSlug: () => constants.FEATURE_BACKUP_ARCHIVE_15,
-		getTitle: () => i18n.translate( '15-day Backup Archive' ),
+		getTitle: () => i18n.translate( '15-day backup archive' ),
 		getDescription: () =>
 			i18n.translate( 'Browse or restore any backup made within the past 15 days.' ),
 	},
 	[ constants.FEATURE_BACKUP_ARCHIVE_UNLIMITED ]: {
 		getSlug: () => constants.FEATURE_BACKUP_ARCHIVE_UNLIMITED,
-		getTitle: () => i18n.translate( 'Unlimited Backup Archive' ),
+		getTitle: () => i18n.translate( 'Unlimited backup archive' ),
 		getDescription: () =>
 			i18n.translate( 'Browse or restore any backup made since you activated the service.' ),
 	},
 	[ constants.FEATURE_BACKUP_STORAGE_SPACE_UNLIMITED ]: {
 		getSlug: () => constants.FEATURE_BACKUP_STORAGE_SPACE_UNLIMITED,
-		getTitle: () => i18n.translate( 'Unlimited Backup Storage Space' ),
+		getTitle: () => i18n.translate( 'Unlimited backup storage space' ),
```

Comment:
:ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.